client/state/reader/lists/reducer.js

```diff
@@ -29,6 +29,8 @@ import {
 import { combineReducers, withSchemaValidation } from 'calypso/state/utils';
 import { itemsSchema, subscriptionsSchema, updatedListsSchema, errorsSchema } from './schema';
 
+const union = ( ...arrays ) => [ ...new Set( [].concat( ...arrays ) ) ];
+
```

Comment:
Looking at the `missingLists` reducer's `READER_LIST_REQUEST_FAILURE` handler, I think the `union` operation always does a `.push`. The new item is an unique object instance that's not going to be `===` to any existing item.

Reply:
Agreed, but that's exactly how Lodash's `union` behaves, too, so I'm inclined to keep it as-is. I'm happy to follow-up with it in another PR, if you think that makes sense. I'd say checking additionally for both of the `action.owner` and `action.slug` should do the job, what do you think?